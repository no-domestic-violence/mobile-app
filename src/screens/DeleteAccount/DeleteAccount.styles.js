import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
  userSettingsContainer: {
    alignItems: 'flex-start',
  },
  arrow: {
    alignSelf: 'flex-start',
    marginTop: 50,
    marginLeft: 10,
  },
  button: {
    backgroundColor: 'red',
    fontWeight: '800',
  },
  view: {
    justifyContent: 'center',
    alignItems: 'center',
    width: '100%',
  },
  header: {
    fontSize: 20,
    fontWeight: '600',
    alignSelf: 'flex-start',
    marginLeft: 30,
    marginBottom: 40,
    marginTop: 40,
  },
  text: {
    fontSize: 16,
    color: '#000',
    alignSelf: 'flex-start',
    marginLeft: 30,
  },
});
