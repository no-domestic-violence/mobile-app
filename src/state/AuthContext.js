import AsyncStorage from '@react-native-async-storage/async-storage';
import createAppContext from './CreateAppContext';
import appApiClient from '../api/appApiClient';

const authReducer = (state, action) => {
  switch (action.type) {
    case 'SIGNUP_ERROR':
    case 'LOGIN_ERROR':
      return {
        ...state,
        errorMessage: action.payload,
      };
    case 'SIGNUP_SUCCESS':
    case 'LOGIN_SUCCESS':
      return {
        ...state,
        token: action.payload.token,
        username: action.payload.user.username,
        errorMessage: '',
      };
    case 'CHANGE_PASSWORD_SUCCESS':
      return {
        ...state,
        successMessage: action.payload,
      };
    case 'CHANGE_PASSWORD_ERROR':
      return {
        ...state,
        errorMessage: action.payload,
      };
    case 'AUTH_SUCCESS':
      return {
        ...state,
        token: action.payload.token,
        username: action.payload.username,
      };
    case 'LOGOUT':
      return {
        token: '',
        username: '',
      };
    case 'DELETE_ACCOUNT':
      return {
        token: '',
        username: '',
        successMessage: action.payload.data,
      };
    case 'REMOVE_ERRORS':
      return {
        ...state,
        errorMessage: '',
      };
    case 'REMOVE_MESSAGES':
      return {
        ...state,
        successMessage: '',
      };
    default:
      return state;
  }
};

const signup = (dispatch) => async ({ email, password, username }) => {
  try {
    const response = await appApiClient.post('/signup', {
      email,
      password,
      username,
    });

    await AsyncStorage.setItem('token', response.data.token);
    await AsyncStorage.setItem('username', response.data.user.username);

    dispatch({ type: 'SIGNUP_SUCCESS', payload: response.data });
  } catch (error) {
    dispatch({
      type: 'SIGNUP_ERROR',
      payload: "Something went wrong'( Try again",
    });
  }
};

const login = (dispatch) => async ({ email, password }) => {
  try {
    const response = await appApiClient.post('/login', { email, password });
    await AsyncStorage.setItem('token', response.data.token);
    await AsyncStorage.setItem('username', response.data.user.username);
    dispatch({ type: 'LOGIN_SUCCESS', payload: response.data });
  } catch (error) {
    dispatch({
      type: 'LOGIN_ERROR',
      payload: 'Are you sure about password and email?',
    });
  }
};

const authentication = (dispatch) => async () => {
  const token = await AsyncStorage.getItem('token');
  const username = await AsyncStorage.getItem('username');
  const user = { token, username };
  if (token) {
    dispatch({ type: 'AUTH_SUCCESS', payload: user });
  }
};
const removeErrors = (dispatch) => () => {
  dispatch({ type: 'REMOVE_ERRORS' });
};

const removeMessages = (dispatch) => () => {
  dispatch({ type: 'REMOVE_MESSAGES' });
};

const changePassword = (dispatch) => async ({
  email,
  password,
  oldPassword,
}) => {
  try {
    const response = await appApiClient.post('/changePassword', {
      email,
      oldPassword,
      password,
    });
    removeErrors();
    dispatch({ type: 'CHANGE_PASSWORD_SUCCESS', payload: response.data });
  } catch (error) {
    dispatch({
      type: 'CHANGE_PASSWORD_ERROR',
      payload: error.response.data.message,
    });
  }
};

const signout = (dispatch) => async () => {
  await AsyncStorage.removeItem('token');
  dispatch({ type: 'LOGOUT' });
};

const deleteAccount = (dispatch) => async ({ username }) => {
  try {
    // TODO: fix sending params so it is more readable
    const response = await appApiClient.delete('/deleteUser', {
      params: { username },
    });
    await AsyncStorage.removeItem('token');
    await AsyncStorage.removeItem('username');
    dispatch({ type: 'DELETE_ACCOUNT', payload: response.data });
  } catch (error) {
    dispatch({
      type: 'CHANGE_PASSWORD_ERROR',
      payload: error.response.data.message,
    });
  }
};

export const { Provider, Context } = createAppContext(
  authReducer,
  {
    signup,
    login,
    signout,
    removeErrors,
    removeMessages,
    authentication,
    changePassword,
    deleteAccount,
  },
  { isLoggedIn: false, errorMessage: '' }
);
