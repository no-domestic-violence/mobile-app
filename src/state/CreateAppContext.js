import React, { useReducer } from 'react';

// boilerplate for creating Context and Provider
// does not to be changed
export default (reducer, actions, defaultValue) => {
  const Context = React.createContext();

  const Provider = ({ children }) => {
    const [state, dispatch] = useReducer(reducer, defaultValue);

    //actions is object for ex {updateCurrentLocation: (dispatch)=> (location) => { dispatch({ type: 'UPDATE_LOCATION', payload: location });}
    const boundActions = {};
    //we need to loop through each action in object and for every key(which is function) we call it with dispatch argument and it gives back function. this function we pass down to all children like ...boundActions
    for (const key in actions) {
      boundActions[key] = actions[key](dispatch);
    }
    return (
      <Context.Provider value={{ state, ...boundActions }}>
        {children}
      </Context.Provider>
    );
  };
  return { Context, Provider };
};
