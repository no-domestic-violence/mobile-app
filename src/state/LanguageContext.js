import React, { createContext, useState } from 'react';
import AsyncStorage from '@react-native-async-storage/async-storage';
import i18n from 'services/i18n';
import PropTypes from 'prop-types';
import { DEFAULT_LANGUAGE } from '../services/i18n/language-detector';

const APP_LANGUAGE = 'lang';

const availableLanguages = [
  { lang: 'en', name: 'English' },
  { lang: 'de', name: 'German' },
];

export const LanguageContext = createContext({
  availableLanguages,
  appLanguage: DEFAULT_LANGUAGE,
  setAppLanguage: () => {},
  initializeAppLanguage: () => {},
});

export const LanguageProvider = ({ children }) => {
  const [appLanguage, setAppLanguage] = useState(DEFAULT_LANGUAGE);

  const setLanguage = async (language) => {
    i18n.changeLanguage(language);
    setAppLanguage(language);
    AsyncStorage.setItem(APP_LANGUAGE, language);
  };

  const initializeAppLanguage = async () => {
    const currentLanguage = await AsyncStorage.getItem(APP_LANGUAGE);
    if (!currentLanguage) {
      setLanguage(DEFAULT_LANGUAGE);
    } else {
      setLanguage(currentLanguage);
    }
  };

  return (
    <LanguageContext.Provider
      value={{
        availableLanguages,
        setAppLanguage: setLanguage,
        appLanguage,
        initializeAppLanguage,
      }}>
      {children}
    </LanguageContext.Provider>
  );
};
LanguageProvider.propTypes = {
  children: PropTypes.node.isRequired,
};
