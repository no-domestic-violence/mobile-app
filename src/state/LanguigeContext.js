//TODO: it did not work, to figure out if we need or we can just you provider for everything


// import React, { createContext } from 'react';
// import { I18nextProvider } from 'react-i18next';
// import i18next from 'i18next';

// const LanguigeContext = createContext();

// function LanguigeProvider(props) {

//   return (
//     <LanguigeContext.Provider>
//       <I18nextProvider i18n={i18next}>{props.children}</I18nextProvider>
//    </LanguigeContext.Provider>
//   );
// }

// export {LanguigeProvider, LanguigeContext};

