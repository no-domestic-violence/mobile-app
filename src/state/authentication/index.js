import createAppContext from 'helpers/createAppContext';
import {
  signup,
  login,
  signout,
  removeErrors,
  removeMessages,
  authentication,
  changePassword,
  deleteAccount,
  setAlreadyLaunchedValue,
} from './authentication.actions';
import authReducer from './authentication.reducer';

export const { Provider, Context } = createAppContext(
  authReducer,
  {
    signup,
    login,
    signout,
    removeErrors,
    removeMessages,
    authentication,
    changePassword,
    deleteAccount,
    setAlreadyLaunchedValue,
  },
  { isLoggedIn: false, errorMessage: '', isFirstLaunch: null }
);
