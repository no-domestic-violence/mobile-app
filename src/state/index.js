import React, {createContext, useState, useMemo} from 'react';

const AppContext = createContext();

function ContextProvider(props) {
  //just an example of context usage
  const [userToken, setUserToken] = useState('');
  const authContext = useMemo(() => {
    return {
      logIn: () => {
        setUserToken('tokenishere');
      },
      signUp: () => {
        setUserToken('tokenishere');
      },
      signOut: () => {
        setUserToken('');
      },
    };
  });

  return (
    <AppContext.Provider value={{authContext, userToken}}>
      {props.children}
    </AppContext.Provider>
  );
}

export {ContextProvider, AppContext};
