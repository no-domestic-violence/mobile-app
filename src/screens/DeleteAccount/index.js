import React, { useContext, useState, useEffect } from 'react';
import {
  View,
  Text,
  StyleSheet,
  TouchableOpacity,
} from 'react-native';
import { Context as AuthContext } from '../../state/AuthContext';

export default function SignUpScreen({ navigation }) {
  const { state, deleteAccount } = useContext(AuthContext);
  const username = state.username;

  const handleDeleteAccount = () => {
    deleteAccount({ username });
  };
  return (
    <View style={styles.view}>
      <Text style={styles.header}>Delete Account</Text>
      <Text style={styles.text}>
        Are you sure that you want to delete account??
      </Text>
      <TouchableOpacity onPress={() => handleDeleteAccount()}>
        <Text style={styles.button}>Delete Account</Text>
      </TouchableOpacity>
      {state.errorMessage ? (
        <Text style={styles.textError}>{state.errorMessage}</Text>
      ) : null}
      {state.successMessage ? (
        <Text style={styles.textSuccess}>{state.successMessage}</Text>
      ) : null}
    </View>
  );
}

const styles = StyleSheet.create({
  input: {
    width: 350,
    height: 55,
    backgroundColor: '#fff',
    margin: 10,
    padding: 8,
    color: '#000',
    borderRadius: 5,
    fontSize: 18,
    fontWeight: '500',
  },
  view: {
    backgroundColor: '#cadeee',
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    height: '100%',
  },
  button: {
    backgroundColor: 'red',
    paddingLeft: 30,
    paddingRight: 30,
    paddingTop: 5,
    paddingBottom: 5,
    borderRadius: 12,
    color: 'white',
    fontSize: 14,
    overflow: 'hidden',
    textAlign: 'center',
    marginTop: 40,
    fontWeight: '600',
  },
  header: {
    fontSize: 35,
    fontWeight: '600',
    alignSelf: 'flex-start',
    marginLeft: 30,
    marginBottom: 40,
  },
  text: {
    fontSize: 14,
    color: '#000',
    alignSelf: 'flex-start',
    marginLeft: 30,
  },
  textError: {
    marginTop: 20,
    color: 'darkred',
  },
  textSuccess: {
    marginTop: 20,
    color: 'darkgreen',
  },
});
