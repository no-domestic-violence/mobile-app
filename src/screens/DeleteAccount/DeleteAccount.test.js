import React from 'react';
import { render } from '@testing-library/react-native';
import { AuthProvider } from 'state/';
import DeleteAccount from './index';

describe('Delete Account screen', () => {
  // TODO: fix passing username
  const component = (
    <AuthProvider>
      <DeleteAccount />
    </AuthProvider>
  );
  it('should match snapshot', () => {
    const result = render(component).toJSON();
    expect(result).toMatchSnapshot();
  });
});
