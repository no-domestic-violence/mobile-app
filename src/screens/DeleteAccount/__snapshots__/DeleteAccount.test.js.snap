// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Delete Account screen should match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#cadeee",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
      Object {
        "alignItems": "flex-start",
      },
    ]
  }
>
  <View>
    <svg
      data-file-name="SvgUserSettings"
      style={
        Object {
          "position": "absolute",
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginLeft": 10,
          "marginTop": 10,
        }
      }
    >
      <
        color="grey"
        icon={
          Object {
            "icon": Array [
              496,
              512,
              Array [],
              "f2bd",
              "M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm0 96c48.6 0 88 39.4 88 88s-39.4 88-88 88-88-39.4-88-88 39.4-88 88-88zm0 344c-58.7 0-111.3-26.6-146.5-68.2 18.8-35.4 55.6-59.8 98.5-59.8 2.4 0 4.8.4 7.1 1.1 13 4.2 26.6 6.9 40.9 6.9 14.3 0 28-2.7 40.9-6.9 2.3-.7 4.7-1.1 7.1-1.1 42.9 0 79.7 24.4 98.5 59.8C359.3 421.4 306.7 448 248 448z",
            ],
            "iconName": "user-circle",
            "prefix": "fas",
          }
        }
        size={80}
      />
      <Text
        style={
          Object {
            "fontSize": 28,
            "marginLeft": 15,
            "textTransform": "capitalize",
          }
        }
      >
        celeste
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <
      color="#000"
      icon={
        Object {
          "icon": Array [
            256,
            512,
            Array [],
            "f104",
            "M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z",
          ],
          "iconName": "angle-left",
          "prefix": "fas",
        }
      }
      onPress={[Function]}
      size={40}
      style={
        Object {
          "alignSelf": "flex-start",
          "marginLeft": 10,
          "marginTop": 50,
        }
      }
      testID="backBtn"
    />
    <Text
      style={
        Object {
          "alignSelf": "flex-start",
          "fontSize": 20,
          "fontWeight": "600",
          "marginBottom": 40,
          "marginLeft": 30,
          "marginTop": 40,
        }
      }
    >
      Delete Account
    </Text>
    <Text
      style={
        Object {
          "alignSelf": "flex-start",
          "color": "#000",
          "fontSize": 16,
          "marginLeft": 30,
        }
      }
    >
      Are you certain you want to delete your account?
    </Text>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#415889",
          "borderRadius": 20,
          "fontWeight": "600",
          "marginTop": 40,
          "opacity": 1,
          "overflow": "hidden",
        }
      }
      testID="deleteAccount"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontSize": 14,
              "fontWeight": "600",
              "overflow": "hidden",
              "paddingBottom": 10,
              "paddingLeft": 50,
              "paddingRight": 50,
              "paddingTop": 10,
              "textAlign": "center",
            },
            Object {
              "backgroundColor": "red",
              "fontWeight": "800",
            },
          ]
        }
      >
        Delete Account
      </Text>
    </View>
  </View>
</View>
`;
